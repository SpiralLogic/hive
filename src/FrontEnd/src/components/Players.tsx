import '../css/players.css'
import { FunctionComponent, h } from 'preact';
import { Players } from '../domain';
import GameTile from './GameTile';
import Player from './Player';

const Players: FunctionComponent<{ players: Players }> = (props) => {
  const { players } = props;
  return (
    <aside className="players">
      {players.map((player) => (
        <Player name={player.name} id={player.id} show={!!player.tiles.length}>
          {player.tiles.map((tile) => (
            <GameTile key={tile.id} {...tile} />
          ))}
        </Player>
      ))}
    </aside>
  );
};

Players.displayName = 'Players';
export default Players;
