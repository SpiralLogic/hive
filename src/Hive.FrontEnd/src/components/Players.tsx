import '../css/players.css';

import { FunctionComponent } from 'preact';

import { PlayerId } from '../domain';
import GameTile from './GameTile';
import Player from './Player';
import { useGameState } from '../services/signals';

const Players: FunctionComponent<{ currentPlayer: PlayerId }> = (properties) => {
  const { currentPlayer } = properties;
  const { players } = useGameState();
  return (
    <aside className="players" aria-label={'PLayer pieces not yet on the board'}>
      {players.value.map((player) => (
        <Player key={`${player.id}`} name={player.name} id={player.id}>
          {player.tiles
            .map((tile) => ({
              ...tile,
              moves: undefined,
            }))
            .map((tile) => (
              <GameTile currentPlayer={currentPlayer} key={`${tile.id}`} {...tile} />
            ))}
        </Player>
      ))}
    </aside>
  );
};

Players.displayName = 'Players';
export default Players;
