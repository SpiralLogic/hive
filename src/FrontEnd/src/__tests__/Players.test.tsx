import { h } from 'preact';
import { renderElement } from './helpers';
import Players from '../components/Players';

describe('PlayerList Tests', () => {
  const ant = { id: 1, playerId: 1, creature: 'ant', moves: [{ q: 1, r: 1 }] };
  const fly = { id: 2, playerId: 0, creature: 'fly', moves: [] };

  const players = [
    { id: 1, name: 'Player 1', tiles: [ant, fly, fly] },
    { id: 2, name: 'Player 2', tiles: [ant] },
  ];

  let playerList: HTMLElement;
  beforeEach(() => {
    global.window.history.replaceState({}, global.document.title, `/game/33/1`);
    playerList = renderElement(<Players players={players} currentPlayer={0} />);
  });

  test('to have class', () => {
    expect(playerList).toHaveClass('players');
  });

  test('players are rendered', () => {
    expect(playerList.getElementsByClassName('player')).toHaveLength(2);
  });

  describe('PlayerList snapshot', () => {
    test('matches current snapshot', () => {
      expect(playerList).toMatchSnapshot();
    });
  });
});
