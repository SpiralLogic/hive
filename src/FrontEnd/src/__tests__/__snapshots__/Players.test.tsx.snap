// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerList Tests snapshot 1`] = `
<aside
  class="players"
>
  <div
    class="player player1"
    title="Player 1"
  >
    <div
      class="name"
    >
      Player 1
    </div>
    <div
      class="tiles"
    >
      <div
        class="player1 hex tile creature"
        draggable="true"
        tabindex="0"
        title="ant"
      >
        <svg
          viewBox="0 0 100 100"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            href="#hex"
          />
          <use
            class="creature ant"
            href="#ant"
          />
        </svg>
      </div>
      <div
        class="player0 hex tile creature"
        title="fly"
      >
        <svg
          viewBox="0 0 100 100"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            href="#hex"
          />
          <use
            class="creature fly"
            href="#fly"
          />
        </svg>
      </div>
      <div
        class="player0 hex tile creature"
        title="fly"
      >
        <svg
          viewBox="0 0 100 100"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            href="#hex"
          />
          <use
            class="creature fly"
            href="#fly"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    class="player player2"
    title="Player 2"
  >
    <div
      class="name"
    >
      Player 2
    </div>
    <div
      class="tiles"
    >
      <div
        class="player1 hex tile creature"
        draggable="true"
        tabindex="0"
        title="ant"
      >
        <svg
          viewBox="0 0 100 100"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            href="#hex"
          />
          <use
            class="creature ant"
            href="#ant"
          />
        </svg>
      </div>
    </div>
  </div>
</aside>
`;
